### ISSUE TITLE
Issues with OpenSSH 6.8 with specific Config

### ISSUE THREAD

**Issue by adur1990** [paramiko/paramiko#526]
My Hoster changed SSH to OpenSSH 6.8 with following config:

```
KexAlgorithms:
curve25519-sha256@libssh.org
diffie-hellman-group-exchange-sha256
diffie-hellman-group14-sha1

Ciphers:
chacha20-poly1305@openssh.com
aes256-gcm@openssh.com
aes128-gcm@openssh.com
aes256-ctr,aes192-ctr
aes128-ctr,aes256-cbc
aes192-cbc,aes128-cbc

MACs:
hmac-sha2-512-etm@openssh.com
hmac-sha2-256-etm@openssh.com
hmac-ripemd160-etm@openssh.com
umac-128-etm@openssh.com
hmac-sha2-512
hmac-sha2-256
hmac-ripemd160
umac-128@openssh.com
```

Connection is not possible any more.

It produces the following Exception:
```
Incompatible ssh server (no acceptable macs)
```

**Comment by xchs** [comment]
I have been having the same problem (and supposedly also with the same hoster :wink:).

**Comment by i42n** [comment]
Please add support for this soon. I am also running into this error.

**Comment by eazyliving** [comment]
Same here. I'd send beer, if it helps ;-)

**Comment by Russell-Jones** [comment]
See https://github.com/paramiko/paramiko/pull/356

**Comment by bitprophet** [comment]
Yup this looks like #356, thanks @Russell-Jones (and all who reported! please follow #356 for updates, will be wanting testing once merged).

**Comment by bitprophet** [comment]
This should be fixed now that work based on #356 is merged to master. Will be released publicly sometime this week.
